### PR TITLE
[5.0] Fix the submenu modules

### PR DESCRIPTION
**Pull request by laoneo**
Pull Request for Issue #41414.

### Summary of Changes
Uses the identity from the app instead of from $this. Adds the respective system test as well.

### Testing Instructions
Open /administrator/index.php?option=com_cpanel&view=cpanel&dashboard=system.

### Actual result BEFORE applying this Pull Request
Error.

### Expected result AFTER applying this Pull Request
No error.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: tests/System/integration/administrator/components/com_cpanel/CPanel.cy.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+describe('Test in backend that the cpanel system dashboard', () => {
```
Comment (brianteeman) [inline tests/System/integration/administrator/components/com_cpanel/CPanel.cy.js:1]: for completeness I would (also) test one of the other dashboards that includes modules as well such as users dashboard

File: administrator/modules/mod_submenu/tmpl/default.php [hunk 2]
```diff
@@ -14,7 +14,7 @@
 use Joomla\CMS\Language\Text;
 use Joomla\CMS\Router\Route;
 
-$user = $this->getCurrentUser();
+$user = $app->getIdentity();
```
Comment (brianteeman) [inline administrator/modules/mod_submenu/tmpl/default.php:17]: as $app is not defined in that file would it not be better to use the original code??
`$user = \Joomla\CMS\Factory::getApplication()->getIdentity();`

Comment (laoneo) [reply administrator/modules/mod_submenu/tmpl/default.php:17]: $app is widely used in modules, see [here](https://github.com/joomla/joomla-cms/blob/5.0-dev/administrator/modules/mod_feed/tmpl/default.php#L27). So it is safe to use...

File: tests/System/integration/administrator/components/com_cpanel/CPanel.cy.js (hunk 1 above)
Comment (laoneo) [reply tests/System/integration/administrator/components/com_cpanel/CPanel.cy.js:1]: We can do that later in another pr.

File: administrator/modules/mod_submenu/tmpl/default.php (hunk 2 above)
Comment (brianteeman) [reply administrator/modules/mod_submenu/tmpl/default.php:17]: but is it good practice?

**Comment by heelc29** [comment]
I have tested this item :white_check_mark: successfully on db3847007c46b1f6f21be91a0575fa6166430f21<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41416">issues.joomla.org/tracker/joomla-cms/41416</a>.</sub>

File: administrator/modules/mod_submenu/tmpl/default.php (hunk 2 above)
Comment (laoneo) [reply administrator/modules/mod_submenu/tmpl/default.php:17]: Better than using Factory...

Comment (brianteeman) [reply administrator/modules/mod_submenu/tmpl/default.php:17]: wtf are you talking about $app is a factory

**Comment by brianteeman** [comment]
I do not agree with the fix in this PR

**Comment by Fedik** [comment]
Brian, use `$app` in the module layout is okay, it is defined. Same as `$params`, `$module` and some other stuff.
It may sounds confusing but it works like that long time, probably since 1.7

**Comment by brianteeman** [comment]
still dont have to like it

**Comment by HLeithner** [comment]
Thanks

**Pull request merged**